src/codec/builtin/BigDecimalCodec.ts

```diff
@@ -11,7 +11,7 @@ export class BigDecimalCodec {
         const contentSize = FixSizedTypesCodec.decodeInt(buffer, 0);
         let body = buffer.slice(BitsUtil.INT_SIZE_IN_BYTES, BitsUtil.INT_SIZE_IN_BYTES + contentSize);
         const byteArray = new Uint8Array(body);
-        const signBit = body.slice(0, 1).readUInt8(0);
+        const signBit = body.readUInt8(0);
```

Comment:
this is not the sign bit. it is the byte that contains the sign bit. To fix the naming, maybe we can remove this and just do `const isNegative = body.readUInt8(0) > 127;` or `const isNegative = body[0] & 0x80 === 1`

Reply:
rather should be `(body[0] & 0x80) > 0`